Lib/test/test_descr.py

```diff
@@ -3900,6 +3900,31 @@ def __ipow__(self, other):
         a = C()
         a **= 2
 
+    def test_ipow_returns_not_implemented(self):
+        class A:
+            def __ipow__(self, other):
+                return NotImplemented
+        class B:
+            def __rpow__(self, other):
+                return 1
+        a = A()
+        b = B()
+        a **= b
+        self.assertEqual(a, 1)
+
+    def test_no_ipow(self):
+        class A:
+            pass
+
+        class B:
+            def __rpow__(self, other):
+                return 1
+
+        a = A()
```

Comment:
This can just be:

```suggestion
        a = object()
```

Then you don't need the class definition above!

Reply:
Done.